### PR TITLE
Refactor preset structure

### PR DESCRIPTION
**Pull request by hiddely**
Refactors data model to better suit PolyCast' feedback

### CODE REVIEW

File: app/models/Preset.java [hunk 1]
```diff
@@ -37,17 +37,19 @@
...
+    @JsonIgnore
+    @ManyToOne(cascade = CascadeType.REFRESH)
+    public Script script;
 
-    @Constraints.Required
-    public int zoom;
+    /**
+     * Used to link to a real world preset.
+     */
+    public int realPresetId;
```
Comment (jorisoudejans) [inline app/models/Preset.java:47]: Real world preset? Let's not use the 'dummy'/'real' terminology. Better is something like 'linkedCameraPreset'.

File: app/models/Preset.java [hunk 2]
```diff
@@ -37,17 +37,19 @@
...
 
-    @Constraints.Required
-    public int focus;
+    /**
+     * State of this preset.
+     */
+    public Status status = Status.OK;
```
Comment (jorisoudejans) [inline app/models/Preset.java:52]: Cool enum

File: app/models/Preset.java (hunk 1 above)
Comment (hiddely) [inline app/models/Preset.java:47]: I am for the dummy/real term. linkedCameraPreset does not imply any external preset but a local one. 😜

**Comment by hiddely** [comment]
Fixed the tests and added some for the new linking

**Pull request merged**